### PR TITLE
Speaker infra (+ 2 little changes in president)

### PR DESCRIPTION
**Pull request by Spruudis**

### CODE REVIEW

File: 05_iigo/index.tex [hunk 1]
```diff
@@ -510,7 +510,7 @@ \subsubsection{Client-side Overview}
...
+The server implementation of each branch provides the IIGO roles with the framework to use their powers of carrying out the law in the archipelago. The server is responsible for the vast majority of data processing and acts as a bridge between the IIGO roles and the other agents. For example, the President decides on the tax distribution for the given turn, and the server converts this information into communication messages and broadcasts them to the islands. This ensures that the backend 'logistics' of the game, such as the format of messages passed between islands, remain the same between different implementations of each IIGO role. It also reduces the complexity of agents and minimizes the amount of work for each of the team implementing their agents.
```
Comment (ezgimez) [inline 05_iigo/index.tex:518]: again...
minimises ---> minimizes.. 
doesn't matter :D but let's go your way

File: 05_iigo/index.tex [hunk 2]
```diff
@@ -547,11 +547,34 @@ \subsubsection{Server-side}
     \item The required information from agents or other roles is obtained. This information includes \emph{allocation requests}, \emph{private pool resource reports} and \emph{rules} proposed by the islands.
     \item This information is aggregated into a format accepted by the interface of presidential functions (usually in form $map[agent] \xrightarrow[]{} information$).
     \item Presidential function is called only on the currently residing President with aggregated information as inputs.
-    \item The outcome of the presidential function call is processed.
+    \item The outcome of the presidential function call is processed. This includes updating the game state, logging information for monitoring, subtracting action costs.
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
    \item The outcome of the presidential function call is processed. This includes updating the game state, logging information for monitoring and subtracting the relevant action costs.
```

File: 05_iigo/index.tex [hunk 3]
```diff
@@ -547,11 +547,34 @@ \subsubsection{Server-side}
...
     \item The outcome of the action, or lack thereof, is passed to the members of the archipelago.
 \end{enumerate}
 
-\subsubsection{Legislative Branch}
+\subsection{Legislative Branch}
+Similarly to the president the Speaker also has client-side functions which directly map to the Speaker powers described in \ref{sec:legislative}, as well as server-side functions which not only facilitate the implementation of these powers but also handle other functionality, such as the logging needed for monitoring and withdrawal of costs associated with the power.
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
Similarly to the President, the Speaker also has client-side functions which directly map to the Speaker's powers described in Section~\ref{sec:legislative}, as well as server-side functions which not only facilitate the implementation of these powers but also handle other functionality, such as the logging needed for monitoring and withdrawal of costs associated with the power.
```

File: 05_iigo/index.tex [hunk 4]
```diff
@@ -547,11 +547,34 @@ \subsubsection{Server-side}
...
+
+\subsubsection{Client-side}
+The legislative branch functions can be divided into two categories:
+\begin{enumerate}
+\item Speaker power functions:
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
\item Legislative power functions:
```

File: 05_iigo/index.tex [hunk 5]
```diff
@@ -547,11 +547,34 @@ \subsubsection{Server-side}
...
+    \begin{itemize}
+        \item \texttt{DecideAgenda} does not directly map to a power of the speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, they can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
        \item \texttt{DecideAgenda} does not directly map to the power of the Speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, they can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
```
who is 'they' in the last sentence?? can you plz be more explicit :)

Comment (ezgimez) [reply 05_iigo/index.tex]: is it 'the Speaker'

File: 05_iigo/index.tex [hunk 6]
```diff
@@ -513,9 +513,9 @@ \subsubsection{Client-side}
     \end{itemize}
 \item Transfer-of-power functions:
     \begin{itemize}
-        \item \texttt{PaySpeaker} implements the obligation of the President to pay the Speaker its salary. 
+        \item \texttt{PaySpeaker} implements the power of the President to pay the Speaker its salary. 
         \item \texttt{CallSpeakerElection} implements the power of the President to initiate the transfer-of-power for the Speaker
-        \item \texttt{DecideNextSpeaker} allows the President to modify the outcome of the election and choose not to follow its obligation to announce the result of Speaker election.
+        \item \texttt{DecideNextSpeaker} allows the President to modify the outcome of the election and choose not to follow its obligation (if Rule "TODO WHEN MY PR IS MERGED" is in play) to announce the result of Speaker election.
```
Comment (ezgimez) [inline 05_iigo/index.tex]: plz change this later :)

File: 05_iigo/index.tex [hunk 7]
```diff
@@ -528,11 +528,33 @@ \subsubsection{Server-side}
...
+    \end{itemize}
+\item The accountability cycle and transfer-of-power functions:
+    \begin{itemize}
+        \item \texttt{PaySpeaker} implements the power of the Speaker to pay the Judge its salary. 
+        \item \texttt{CallJudgeElection} implements the power of the Speaker to call an election for the transfer-of-power for the Judge
+        \item \texttt{DecideNextJudge} allows the Speaker to modify the outcome of the election, if one is called, and choose not to follow its obligation (if Rule "TODO WHEN PR 4 IS MERGED" is in play) to announce the result of Judge election.
```
Comment (ezgimez) [inline 05_iigo/index.tex]: can you change is TODO plz

File: 05_iigo/index.tex [hunk 8]
```diff
@@ -528,11 +528,33 @@ \subsubsection{Server-side}
...
+    \end{itemize}
+\end{enumerate}
+
+\subsubsection{Server-side}
+The speaker server-side is responsible for enabling the information flow, maintaining simulation concurrency concerning action costs, updating the game state, and logging actions for monitoring. Notable functions are \texttt{updateRules}, which implements the concurrency checks and logic for updating the rules in play, and \texttt{RunVote}, which interacts with the voting system.
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
The server-side of the legislative branch is responsible for enabling the information flow, maintaining simulation concurrency concerning action costs, updating the game state and logging actions for monitoring. Notable functions are \texttt{updateRules}, which implements the concurrency checks and logic for updating the rules in play, and \texttt{RunVote}, which interacts with the voting system.
```

File: 05_iigo/index.tex [hunk 9]
```diff
@@ -528,11 +528,33 @@ \subsubsection{Server-side}
...
+        \item \texttt{DecideAnnouncement} enables the institutional power of declaring a result of a vote on rules. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and the result of the vote held. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker along with the result returned by the Speaker is the information that is processed by the server and broadcasted to all the agents if the agent has chosen to make a declaration as indicated by the boolean returned by the Speaker.
+    \end{itemize}
+\item The accountability cycle and transfer-of-power functions:
+    \begin{itemize}
+        \item \texttt{PaySpeaker} implements the power of the Speaker to pay the Judge its salary. 
+        \item \texttt{CallJudgeElection} implements the power of the Speaker to call an election for the transfer-of-power for the Judge
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
        \item \texttt{CallJudgeElection} implements the power of the Speaker to call an election for the transfer-of-power for the Judge.
```

File: 05_iigo/index.tex (hunk 7 above)
Comment (ezgimez) [reply 05_iigo/index.tex]: reference rule properly plz..
`Rule~\ref{rule:XX}` or smth like that should work

File: 05_iigo/index.tex [hunk 10]
```diff
@@ -528,11 +528,33 @@ \subsubsection{Server-side}
...
+
+\subsubsection{Client-side}
+The legislative branch functions can be divided into two categories:
+\begin{enumerate}
+\item Speaker power functions:
+    \begin{itemize}
+        \item \texttt{DecideAgenda} does not directly map to a power of the speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, they can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
+        \item \texttt{DecideVote} enables the institutional power of calling a vote and deciding the participating islands. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and a list of all the alive island IDs. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker is the rule that is passed to agents returned by the Speaker if a vote is to be conducted as indicated by the boolean returned by the Speaker.
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
        \item \texttt{DecideVote} enables the institutional power of calling a vote and deciding which islands participate to this engagement. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and a list of all the alive island IDs. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker is the rule that is passed to agents returned by the Speaker if a vote is to be conducted as indicated by the boolean returned by the Speaker.
```

File: 05_iigo/index.tex [hunk 11]
```diff
@@ -528,11 +528,33 @@ \subsubsection{Server-side}
...
+        \item \texttt{DecideAnnouncement} enables the institutional power of declaring a result of a vote on rules. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and the result of the vote held. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker along with the result returned by the Speaker is the information that is processed by the server and broadcasted to all the agents if the agent has chosen to make a declaration as indicated by the boolean returned by the Speaker.
```
Comment (ezgimez) [inline 05_iigo/index.tex]: result of a vote on rules/??
 can you revise this plz , not very understandable

**Review by ezgimez — COMMENTED** [review]
minor suggestions

File: 05_iigo/index.tex (hunk 5 above)
Comment (Spruudis) [reply 05_iigo/index.tex]:
```suggestion
        \item \texttt{DecideAgenda} does not directly map to any power of the Speaker, but it does aid the agents with keeping track of the rule propagated through the legislative branch functions. By changing the rule given to the agent in \texttt{DecideAgenda}, which is the same rule returned by the President in \texttt{PickRuleToVote}, agents can alter the rule passed to them in \texttt{DecideVote} and \texttt{DecideAnnouncement}.
```

**Review by ezgimez — APPROVED** [review]
lgtm, let's merge

( I don't wanna waste time on some petty conflicts anymore :) )

File: 05_iigo/index.tex [hunk 12]
```diff
@@ -570,11 +570,33 @@ \subsection{Server-side Implementation}
...
+        \item \texttt{DecideAnnouncement} enables the institutional power of declaring a result of a vote on rules. The information passed to the Speaker is the current rule in the agenda (as returned by \texttt{DecideAgenda}) and the result of the vote held. The Speaker returns the same type of information along with a boolean indicating whether the agent has chosen to act on this power. The rule returned by the Speaker along with the result returned by the Speaker is the information that is processed by the server and broadcasted to all the agents. This does not occur if the agent has chosen not to make a declaration as indicated by the returned boolean.
+    \end{itemize}
+\item The accountability cycle and transfer-of-power functions:
```
Comment (ezgimez) [inline 05_iigo/index.tex:589]:
```suggestion
\item The accountability cycle and transfer-of-power functions are:
```

File: 05_iigo/index.tex [hunk 13]
```diff
@@ -570,11 +570,33 @@ \subsection{Server-side Implementation}
...
+    \begin{itemize}
+        \item \texttt{PaySpeaker} implements the power of the Speaker to pay the Judge its salary. 
+        \item \texttt{CallJudgeElection} implements the power of the Speaker to call an election for the transfer-of-power for the Judge
```
Comment (ezgimez) [inline 05_iigo/index.tex]:
```suggestion
        \item \texttt{CallJudgeElection} implements the power of the Speaker to call an election for the transfer-of-power for the Judge.
```

**Pull request merged**